### ISSUE TITLE
move or remove plugins.rst?

### ISSUE THREAD

**Issue by Gustry** [qgis/QGIS-Documentation#3819]
## Description

IMHO the `plugins.rst` page is complicated for normal Python developers.
I'm not sure that these snippets are still working on QGIS 3 and even myself, I'm not sure about the real use case of plugin layers. I guess, 99% of python developers works with "normal" layers. Sorry if I missed the real use case.

https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/plugins/pluginlayer.html

Should we move it elsewhere? IMHO it breaks the workflow of learning how to write a plugin:
https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/plugins/index.html#developing-python-plugins

And the name might be very confusing for beginners as it contains the word `plugin`.

Or should we remove this page? It is still not updated for QGIS3.

@alexbruy You seem to be the author of that page. What do you think about this page?

**Comment by DelazJ** [comment]
@Gustry indeed, it somehow breaks the reading (and looks too specialized). But it's now up to date with 3.x and under testing.
Maybe should we at least move it to the end of the plugins chapter? @alexbruy @elpaso any idea of a better place? Thanks

**Comment by elpaso** [comment]
IMO that page is too important to be removed, it's the only place I've found where that feature is documented, I had to stop and learn before I could review it.

... and I agree the name "plugin layers" is misleading in the context of python plugin learning, but the cookbook is not a tutorial about writing plugins, it is about PyQGIS in general.

I think we should move it out of the "writing plugins" chapter into its own section.
